livekit-ffi/src/server/requests.rs

```diff
@@ -788,6 +790,65 @@ fn on_flush_sox_resampler(
     }
 }
 
+fn on_perform_rpc(
+    server: &'static FfiServer,
+    request: proto::PerformRpcRequest,
+) -> FfiResult<proto::PerformRpcResponse> {
+    let ffi_participant =
+        server.retrieve_handle::<FfiParticipant>(request.local_participant_handle)?.clone();
+    return ffi_participant.perform_rpc(server, request);
+}
+
+fn on_register_rpc_method(
+    server: &'static FfiServer,
+    request: proto::RegisterRpcMethodRequest,
+) -> FfiResult<proto::RegisterRpcMethodResponse> {
+    let ffi_participant =
+        server.retrieve_handle::<FfiParticipant>(request.local_participant_handle)?.clone();
+    return ffi_participant.register_rpc_method(server, request);
+}
+
+fn on_unregister_rpc_method(
+    server: &'static FfiServer,
+    request: proto::UnregisterRpcMethodRequest,
+) -> FfiResult<proto::UnregisterRpcMethodResponse> {
+    let ffi_participant =
+        server.retrieve_handle::<FfiParticipant>(request.local_participant_handle)?.clone();
+    return ffi_participant.unregister_rpc_method(server, request);
+}
+
+fn on_rpc_method_invocation_response(
+    server: &'static FfiServer,
+    request: proto::RpcMethodInvocationResponseRequest,
+) -> FfiResult<proto::RpcMethodInvocationResponseResponse> {
+    let async_id = server.next_id();
+    let ffi_participant =
+        server.retrieve_handle::<FfiParticipant>(request.local_participant_handle)?.clone();
+
+    let room = ffi_participant.room;
+
+    let handle = server.async_runtime.spawn(async move {
+        let mut error: Option<String> = None;
+
+        if let Some(waiter) = room.take_rpc_method_invocation_waiter(request.invocation_id) {
+            let result = if let Some(error) = request.error.clone() {
+                Err(RpcError { code: error.code, message: error.message, data: error.data })
+            } else {
+                Ok(request.payload.unwrap_or_default())
+            };
+            let _ = waiter.send(result);
+        } else {
+            error = Some("No caller found".to_string());
+        }
+
+        let callback = proto::RpcMethodInvocationResponseCallback { async_id, error };
+        let _ = server.send_event(proto::ffi_event::Message::RpcMethodInvocationResponse(callback));
```

Comment:
same for this message?

Reply:
you're right - i think I convinced myself that the ffi response shouldn't send anything besides async_id but that's not true, just how it worked in many of the examples i saw. i merged the error into the response itself and returned that directly with no callback.